### DISCUSSION TITLE
Apply OpenType Features in IDE’s

### DISCUSSION THREAD

**Discussion by Mark2Mark** [Mark2Mark/gintronic#11] [Q&A]
I want to collect the ways with which certain font features can be enabled/disabled in different IDE’s

Feel free to add your favourite IDE here.

# VSCode

e.g.
```json
"editor.fontLigatures": "'ss01', 'ss02', 'ss03', 'ss04', 'ss05', 'ss06', 'zero', 'onum'"
```
or
```json
    "editor.fontFamily": "Gintronic"    
    "editor.fontLigatures": true,
    "[cpp]": {
        "editor.fontLigatures": "'ss01', 'ss02'",
    },
    "[javascript]": {
        "editor.fontLigatures": "'ss01'",
    }, 
```

[Source](https://github.com/microsoft/vscode/issues/10435). It seems you can apply the settings language based. I find the name `editor.fontLigatures` a little misleading, as ligatures are only a subset of the font features, but it handles the font features as a whole.

> [!NOTE]  
> Please note that Gintronic currently does not (yet) have coding ligatures. See #13

**Comment by ElectronSutra** [comment]
## Kitty Terminal Emulator

The Kitty terminal emulator specifies font variants like so:
```
font_features Gintronic-Regular +ss01 +ss02 +onum
font_features Gintronic-Black   +ss01 +ss02 +onum
```
and so on. Note that Kitty uses the "formal" name of the font for specifying font features, which you can specify in the terminal using the `kitty +list-fonts --psnames` command.

## Wezterm

The Wezterm terminal emulator is configured using Lua, with stylistic sets applied like so:

```
CONFIG = {}

CONFIG.harbuzz_features = { "ss01", "ss02", "onum" }

return CONFIG
```

Or at least, that's how I style it.

## Emacs 28+
As far as I can tell, stylistic set and character variants have to be baked-in to the font to get them to display properly in Emacs. Ligatures (which if I interpret it correctly are also part of the OpenType features, although I have to admit I don't know much about the internals of fonts) are enabled by using the `ligatures.el` package from MELPA, and specifying the character REGEXes to check:
```
(use-package ligature
  [...]
  :config
  (ligature-set-ligatures 'prog-mode '("->", "|>", [...])); enable these ligatures only for code
  (ligature-set-ligatures 't '("www")); This ligature is global
)
```
I'm relatively new at Emacs so I might be mistaken, but this the advice I was most recently able to find.

Thank you for the lovely font ❤️

**Reply by Mark2Mark** [reply]
Thanks so much for your input! I really appreciate it. Glad you like the font.

**Reply by Mark2Mark** [reply]
Also you understand correctly that ligatures are part of the OpenType Features. They are enabled/disabled the same way. However there are different approaches when making the font, where you can specify which ligatures will always show and which will show on user’s discretion.

**Comment by dongz9** [comment]
The above configuration doesn't seem to enable ligature for me. I wonder which font features should I enable?

**Comment by Mark2Mark** [comment]
Hey @dongz9, which ligatures are you achieving to see? The current version of Gintronic does not (yet) have ligatutes. See #13

**Reply by dongz9** [reply]
It will be nice to include ligatures similar to Fira Code, MonoLisa and JetBrains Mono.
